### PR TITLE
Fixed some bugs, added time algorithm, and time in event UI

### PR DESCRIPTION
**Pull request by Mangr7743**
Used the availability tab results to get the time of all users in an event and then combine them and create an algorithm which finds first free space for all users. This is returned to the event object and the update is reflected in the event UI

### CODE REVIEW

**Review by jathu-oai — APPROVED** [review]

**Comment by jathu-oai** [comment]
fixes look good on high level, but waiting for another approval before merging

**Review by z22shaik — APPROVED** [review]
LGTM

**Pull request merged**
